strategies/categorized/src/test/java/org/arquillian/smart/testing/strategies/categorized/CategorizedTestsDetectorTest.java

```diff
@@ -213,6 +208,94 @@ public void should_return_classes_and_methods_without_first_category_when_revers
                 testSelectionWithMethods);
     }
 
+
+    @Test
+    public void should_return_methods_without_excluded_category() {
+        //given
+        categorizedConfig.setCategories(new String[] {Included.class.getSimpleName()});
+        categorizedConfig.setExcludedCategories(new String[] {Excluded.class.getSimpleName()});
+
+        //when
+        Collection<TestSelection> testSelection =
+            new CategorizedTestsDetector(config).selectTestsFromClasses(Collections.singleton(WithExcludedMethodClass.class));
+
+        //then
+        TestSelection testSelectionWithMethods = getTestSelectionWithMethods(WithExcludedMethodClass.class, "testWithoutExcludedCategory");
+
+        assertThat(testSelection).containsTestClassSelectionsExactlyInAnyOrder(testSelectionWithMethods);
+    }
+
+    @Test
+    public void should_return_methods_without_excluded_tag() {
+        //given
+        categorizedConfig.setCategories(new String[] {"first"});
+        categorizedConfig.setExcludedCategories(new String[] {"second","third"});
+
+        //when
+        Collection<TestSelection> testSelection =
+            new CategorizedTestsDetector(config).selectTestsFromClasses(Collections.singleton(WithTaggedMethodsClass.class));
+
+        //then
+        TestSelection testSelectionWithMethods = getTestSelectionWithMethods(WithTaggedMethodsClass.class, "testWithFirstTag");
+
+        assertThat(testSelection).containsTestClassSelectionsExactlyInAnyOrder(testSelectionWithMethods);
+    }
+
+    @Test
+    public void should_return_class_when_no_methods_are_matched_to_be_excluded() {
+        //given
+        categorizedConfig.setCategories(new String[] {FirstCategory.class.getSimpleName()});
+        categorizedConfig.setExcludedCategories(new String[] {Excluded.class.getSimpleName()});
+
+        //when
+        Collection<TestSelection> testSelection =
+            new CategorizedTestsDetector(config).selectTestsFromClasses(Collections.singleton(FirstCategorizedClass.class));
+
+        assertThat(testSelection).containsTestClassSelectionsExactlyInAnyOrder(getTestSelection(FirstCategorizedClass.class));
+    }
+
+    @Test
+    public void should_not_exclude_test_class_when_excluding_single_category_of_many() {
+        //given
+        categorizedConfig.setCategories(new String[] {FirstCategory.class.getSimpleName()});
+        categorizedConfig.setExcludedCategories(new String[] {SecondCategory.class.getSimpleName()});
```

Comment:
The test is called: `should_not_exclude_test_class_when_excluding_single_category_of_many` and you exclude the `SecondCategory` which is contained in `FirstAndSecondCategorizedClass`. I thought that if the class/method has the category that is excluded, then it shouldn't be executed at all. it is also described here: http://maven.apache.org/surefire/maven-surefire-plugin/test-mojo.html#excludedGroups